app/controllers/landings_controller.rb

```diff
@@ -1,28 +1,30 @@
 class LandingsController < PagesController
   before_action :save_form_info, only: %i[index show]
-  before_action :retrieve_landing, only: %i[show create_solicitation]
+  before_action :retrieve_landing, only: %i[show create_solicitation new_solicitation]
 
   def index
     @landings = Rails.cache.fetch('landings', expires_in: 3.minutes) do
       Landing.ordered_for_home.to_a
     end
   end
 
-  def show
+  def show; end
+
+  def new_solicitation
+    @landing_options_slugs = params[:option].nil? ? [] : [params[:option]]
     @solicitation = @landing.solicitations.new
```

Comment:
En fait je me demande si on peut pas se passer complètement de `@landing_options_slugs` en assignant dès le début à la solicitation créée.
```suggestion
    @solicitation = @landing.solicitations.new(landing_options_slugs: [params[:option]].compact)
```

(En fait `@solicitation` a déjà un attribut valué: `@landing.landing_slug` vaut implicitement `@landing.slug` parce qu’on crée un nouvel objet sur la relation. On ne l’utilise pas dans le formulaire, parce qu’on refait `@landing.solicitations.create`, mais on pourrait le mettre dans un `hidden_field` et créer la solicitation avec `Solicitation.create`. Bref.)

Reply:
exact, je fais ça je trouve ça bien